lib/model/port.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * port.js
+ *
+ * Definition for the Port class. Ports are the points on a Node where
+ * connections attach to or from the node. They can be either inputs
+ * or outputs.
+ */
+
+/**
+ * Constructor function.
+ *
+ * @param id - The Port's unique identifier.
+ * @param type - The type of port. Valid options include "input" or "output".
+ * @param direction - cardinal direction (N,S,E,W) in which the port is located on the node.
+ * @param percentage - percentage (from left to right or bottom to top) in which the port is located
+ * 					   on the cardinal direction specified by the direction parameter.
+ */
+var Port = function(id,type,direction,percentage) {
+	this.id = id || 0;
+	this.direction = direction;

```

Comment:
its not, I was just thinking using single character strings which would be case independant, so "N" or "n" for north. we can change that though and make enums for more clarity.